app/status/views.py

```diff
@@ -1,11 +1,36 @@
 import clamd
-from flask import request
+from flask import current_app, request
+from psutil import net_connections, Process
 
 from . import status
 from ..clam import get_clamd_socket
 from dmutils.status import get_app_status, StatusError
 
 
+def get_clamd_status():
+    try:
+        pid = next(
+            conn.pid
+            for conn in net_connections(kind="unix")
+            if conn.laddr == current_app.config["DM_CLAMD_UNIX_SOCKET_PATH"]
+        )
+    except StopIteration:
+        raise StatusError('No connection found matching DM_CLAMD_UNIX_SOCKET_PATH')
+
+    if not pid:
+        # insufficient permissions?
+        raise StatusError("Unable to determine pid of process connected to DM_CLAMD_UNIX_SOCKET_PATH")
+
+    process = Process(pid)
+    process.send_signal(0)
```

Comment:
The signal send is to check the process isn't locked up right? Otherwise just instantiating `Process` with the pid will raise if it doesn't exist

Reply:
not entirely sure what gets raised in this case... wonder what i could use to test this. maybe if i send `socat` a `SIGSTOP` it won't be able to handle the new signal.

Reply:
No of course a "stopped" process is still able to receive signals - it must be. I need to get `socat` stuck in an uninterruptible syscall.

Reply:
Reading about how to deliberately get a process in uninterruptable sleep, most solutions seem to involve writing a misbehaving kernel module. Not going down this road. Safe to say, it will raise some sort of exception that will result in a 500.